desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -178,6 +175,42 @@ DCBUTTON(2)
 DCBUTTON(3)
 DCBUTTON(4)
 
+// Delete DC button slots
+#define DDCBUTTON(num) \
+void DownloadFromDCWidget::DDC##num##Clicked() \
+{ \
+	ui.DC##num->setVisible(false); \
+	ui.DDC##num->setVisible(false); \
+	int dc = num; \
+	switch (dc) { \
+		case 1: \
+			qPrefDiveComputer::set_vendor1(qPrefDiveComputer::vendor2()); \
+			qPrefDiveComputer::set_product1(qPrefDiveComputer::product2()); \
+			qPrefDiveComputer::set_device1(qPrefDiveComputer::device2()); \
+		case 2: \
+			qPrefDiveComputer::set_vendor2(qPrefDiveComputer::vendor3()); \
+			qPrefDiveComputer::set_product2(qPrefDiveComputer::product3()); \
+			qPrefDiveComputer::set_device2(qPrefDiveComputer::device3()); \
+		case 3: \
+			qPrefDiveComputer::set_vendor3(qPrefDiveComputer::vendor4()); \
+			qPrefDiveComputer::set_product3(qPrefDiveComputer::product4()); \
+			qPrefDiveComputer::set_device3(qPrefDiveComputer::device4()); \
+		case 4: \
+			qPrefDiveComputer::set_vendor4(QString()); \
+			qPrefDiveComputer::set_product4(QString()); \
+			qPrefDiveComputer::set_device4(QString()); \
+	} \
+	qPrefDiveComputer::set_vendor(qPrefDiveComputer::vendor1()); \
```

Comment:
Not sure why this we are overwriting these preferences here?

Reply:
I may have this wrong, but my intention was to leave the prefs file as though the deleted computer never existed so they are renumbered.   In other words, if we have computers 1..4 and we delete #2, we'll end up with 1..3 instead of 1,3,4.

Reply:
The shifting around of the 'remembered' dive computers 1...4 makes sense - but I am not sure if setting the one dive computer that is shown when opening the dialog to the remembered dive computer 1 in every case is the right thing to do. I would probably just leave this, and assume that the user will be selecting a different model if this was the deleted dive computer.

Reply:
I was trying to avoid setting the remembered dive computer to one that had been deleted, but I was on the fence about this behavior and I think it may have had a bad side effect if you deleted all but 1 computer.

Reply:
Can you please try to reproduce and document what the side effect was?
I think it is not ideal to change the pre-selected dive computer in cases when it is not the preset that is being deleted - so we should either:
- not change the pre-selected dive computer - if the user deletes this preset and then selects another dive computer to download from this will take care of updating the pre-select;
- only change the pre-selected dive computer if this is the remembered dive computer that was deleted.

Reply:
I didn't look very closely at causation as I agreed with your suggestion to leave the selected dive computer unchanged.   With the code in to change it to dive computer #1, if you deleted all but 1 dive computer the selected dive computer manufacturer, model, and device wasn't changed and clicking on the remaining dive computer did nothing.